src/s3/client.rs

```diff
@@ -180,9 +186,9 @@ impl ClientBuilder {
 #[derive(Clone, Debug, Default)]
 pub struct Client {
     client: reqwest::Client,
-    base_url: BaseUrl,
+    pub base_url: BaseUrl,
     provider: Option<Arc<Box<(dyn Provider + Send + Sync + 'static)>>>,
-    region_map: Arc<DashMap<String, String>>,
+    pub region_map: Arc<DashMap<String, String>>,
```

Comment:
```suggestion
    pub(crate) region_map: Arc<DashMap<String, String>>,
```

This prevents usage by non-minio-rs code.